server/chess.py

```diff
@@ -271,14 +271,19 @@ def validate_king_move(
             if (not rook) or rook.has_moved:
                 return False
             for empty_file in empty_files:
-                if not self.get_piece(rank, empty_file):
+                if self.get_piece(rank, empty_file):
                     return False
-            return not self.hypothetical_check(
-                king.side, (king, rank, file), (rook, rank, rook_end)
-            )
+                elif self.hypothetical_check(
+                        king.side, (king, rank, empty_file)):
+                    return False
```

Comment:
This would also check that the empty tile the rook moved through in queen-side castling was not attacked, which is not necessary. And it won't check that the king's starting position was not attacked. The only two squares we need to check are the one the king starts on and the empty one it goes through. I forgot to do both, and instead checked the one it arrived on which is checked elsewhere.
```suggestion
            # FIXME: Check the king is not moving out of or through check.
```

Reply:
Great, will change that now!